libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -0,0 +1,267 @@
+        return request
+    }
+    
+    /// Execute a create.
+    /// - Parameter objectType: An object type.
+    /// - Parameter fields: Dictionary of fields
+    /// - Parameter completionBlock: The completion block to invoke.
+    func create(_ objectType: String, fields: [String:Any], _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.requestForCreate(withObjectType: objectType, fields: fields, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute resources request
+    /// - Parameter completionBlock: The completion block to invoke.
+    func resources(_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forResources: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute versions request
+    /// - Parameter completionBlock: The completion block to invoke.
+    func apiVersions(_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.requestForVersions()
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute fileRendition request
+    /// - Parameter fileId: An sfdc file identifier.
+    /// - Parameter version: A file version.
+    /// - Parameter type: File type.
+    /// - Parameter page: Page number.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func fileRendition(_ fileId: String, version: String, type: String, page: Int, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forFileRendition: fileId, version: version, renditionType: type, page: UInt(page), apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a searchScopeAndOrder request
+    /// - Parameter completionBlock: The completion block to invoke.
+    func searchScopeAndOrder(_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSearchScopeAndOrder: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a searchResultLayout
+    /// - Parameter objectList: Array of objects.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func searchResultLayout(_ objectList:[String],_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSearchResultLayout: objectList.joined(separator: ","), apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+}
```

Comment:
What about batch / composite and sobject tree?

Reply:
good question. I think I completely missed that.